packages/react-instantsearch-dom-maps/src/withGoogleMaps.tsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+type Omit<T, K> = Pick<T, Exclude<keyof T, K>>;
+type Subtract<T, K> = Omit<T, keyof K>;
+
+// @TODO: Move to GoogleMaps once it's migrated to TypeScript
+export interface GoogleMapsContext {
+  __ais_geo_search__google_maps__: {
+    google: typeof google;
+    instance: google.maps.Map;
+  };
+}
+
+export interface WithGoogleMapsProps {
+  google: typeof google;
+  googleMapsInstance: google.maps.Map;
+}
+
+const withGoogleMaps = <Props extends WithGoogleMapsProps>(
```

Comment:
naming is somewhat inconsistent, but it makes sense

Reply:
You mean `withXXX` vs `connectXXX` ?